### PR TITLE
EtherCat support

### PR DESCRIPTION
**Pull request by speakinghedge**
this is the cleaned up version of broken PR #569

it incorporates all requested changes up to 3da7f3f (the last change requested [here](https://github.com/secdev/scapy/pull/569#pullrequestreview-99155688)).

sorry for the noise.

### CODE REVIEW

**Comment by gpotter2** [comment]
Might be a bit late, but you forgot to configure git :/ you should re-author your commits

**Comment by speakinghedge** [comment]
I definitively need more sleep. Or a new brain. Or both.

**Comment by p-l-** [comment]
I've restarted two failed tests.

**Comment by speakinghedge** [comment]
well - this was a quick merge. thank you :+1:

**Comment by gpotter2** [comment]
Nice job !

File: scapy/contrib/ethercat.py [hunk 1]
```diff
@@ -0,0 +1,647 @@
...
+
+                    elif field_required_bits > octet_bits_left:
+                        # take remaining bits
+                        cur_val |= octet << cur_val_bit_idx
+
+                        cur_val_bit_idx += octet_bits_left
+                        field_required_bits -= octet_bits_left
+                        octet_bits_left = 0
+
+            return s[bf_total_byte_length:], this_val
```
Comment (gpotter2) [inline scapy/contrib/ethercat.py]: Btw, small question: are those LEBitField universal of ethercat specific ?

We needed one of those in layers/bluetooth.py
https://github.com/secdev/scapy/blob/master/scapy/layers/bluetooth.py#L92
And instead have a predissect hack

**Comment by speakinghedge** [comment]
> Btw, small question: are those LEBitField universal of ethercat specific ?

This is not EtherCat specific - but I don't know if I added all of those functions forming the body of a real generic field type (so the honest answer is: works for me).

**Pull request merged**